ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -947,3 +950,27 @@ def _router_bgp_mlag_peer_group(self) -> dict:
 
         router_bgp["address_family_ipv4"] = {"peer_groups": [address_family_ipv4_peer_group]}
         return strip_empties_from_dict(router_bgp)
+
+    def _address_family_peer_groups(self, peer_group_name: str, router_bgp: dict, bgp_peer_group: list, address_family: str = "address_family_ipv4") -> None:
```

Comment:
Please just handle both address families and the global inside this function. Then you can call it once per peer-group under the main logic.